src/typing/flow_js.ml

```diff
@@ -1547,7 +1547,14 @@ let rec flow cx (l,u) trace =
       ->
       (* if inflowing type is literal (thus guaranteed to be
          unaliased), propertywise subtyping is sound *)
-      let lit = (desc_of_reason reason1) = "object literal" in
+      let lit =
+        let desc = (desc_of_reason reason1) in
+        let is_prefix a b =
```

Comment:
Here's a simpler is_prefix:

    let is_prefix a b = String.(
      let n = length b in 
      length a >= n && sub a 0 n = b)